### PR TITLE
Add Web3js Content 

### PR DESCRIPTION
**Pull request by themacexpert**
### Description

Adds Web3Js Content. 

### Checklist

- [x] If this page requires a disclaimer, I have added one
- [ ] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects

### CODE REVIEW

File: builders/interact/ethereum-api/libraries/web3js.md [hunk 1]
```diff
@@ -0,0 +1,498 @@
+---
+title: Send Transactions & Deploy Contracts with Web3.js
+description: Follow this tutorial to learn how to use the Ethereum Web3 JavaScript Library to send transactions and deploy Solidity smart contracts to your Tanssi EVM ContainerChain.
+---
+
+# Web3.js JavaScript Library
+
+## Introduction {: #introduction }
+
+[Web3.js](https://web3js.readthedocs.io/){target=_blank} is a set of libraries that allow developers to interact with Ethereum nodes using HTTP, IPC, or WebSocket protocols with JavaScript. Tanssi EVM ContainerChains have an Ethereum-like API available that is fully compatible with Ethereum-style JSON RPC invocations.  Therefore, developers can leverage this compatibility and use the Web3.js library to interact with a Tanssi EVM ContainerChain node as if they were doing so on Ethereum. For more information on Web3.js, check their [documentation site](https://web3js.readthedocs.io/en/v1.10.0/){target=_blank}.
```
Comment (eshaben) [inline builders/interact/ethereum-api/libraries/web3js.md:10]:
```suggestion
[Web3.js](https://web3js.readthedocs.io/){target=_blank} is a set of libraries that allow developers to interact with Ethereum nodes using HTTP, IPC, or WebSocket protocols with JavaScript. Tanssi EVM ContainerChains have an Ethereum-like API that is fully compatible with Ethereum-style JSON RPC invocations.  Therefore, developers can leverage this compatibility and use the Web3.js library to interact with a Tanssi EVM ContainerChain node as if they were doing so on Ethereum. For more information on Web3.js, check out their [documentation site](https://web3js.readthedocs.io/en/v1.10.0/){target=_blank}.
```

File: builders/interact/ethereum-api/libraries/web3js.md [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+
+In this guide, you'll learn how to use setup the Web3.js library for your Tanssi EVM ContainerChain. Next, to showcase the library in action, you'll use Web3.js library to send a transaction and deploy a contract on a Tanssi EVM ContainerChain running in Tanssi's [Dancebox](XXX){target=_blank} TestNet. This guide can be adapted for your own Tanssi EVM ContainerChain by simply changing the endpoint.
```
Comment (eshaben) [inline builders/interact/ethereum-api/libraries/web3js.md:12]:
```suggestion
In this guide, you'll learn how to set up the Web3.js library for your Tanssi EVM ContainerChain. Next, to showcase the library in action, you'll use the Web3.js library to send a transaction and deploy a contract on a Tanssi EVM ContainerChain running in Tanssi's [Dancebox](XXX){target=_blank} TestNet. This guide can be adapted for your own Tanssi EVM ContainerChain by simply changing the endpoint.
```

File: builders/interact/ethereum-api/libraries/web3js.md [hunk 3]
```diff
@@ -0,0 +1,498 @@
...
+
+## Checking Prerequisites {: #checking-prerequisites } 
+
+For the examples in this guide, you will need to have the following:
+
+ - An account with funds in the Tanssi EVM ContainerChain you are testing with
+
+## Create a JavaScript Project {: #create-a-javascript-project }
+
+To get started, you can create a directory to store all of the files you'll be creating throughout this guide:
```
Comment (eshaben) [inline builders/interact/ethereum-api/libraries/web3js.md:24]: I believe this changed in the moonbeam-docs repo to include initializing the project. Please update this section to match

ref: https://github.com/moonbeam-foundation/moonbeam-docs/blob/master/builders/build/eth-api/libraries/web3js.md?plain=1#L28

File: builders/interact/ethereum-api/libraries/web3js.md [hunk 4]
```diff
@@ -0,0 +1,498 @@
...
+=== "yarn"
+
+    ```bash
+    yarn add ethers solc@0.8.0
+    ```
+
+
+## Setup Web3.js with Tanssi {: #setup-web3-with-tanssi }
+
+You can configure Web3.js to work with a Tanssi EVM ContainerChain running in Tanssi's [Dancebox](XXX){target=_blank} TestNet, or your own Tanssi EVM ContainerChain by simply changing the endpoint.
```
Comment (eshaben) [inline builders/interact/ethereum-api/libraries/web3js.md:47]:
```suggestion
You can configure Web3.js to work with a Tanssi EVM ContainerChain running in Tanssi's [Dancebox](XXX){target=_blank} TestNet or your own Tanssi EVM ContainerChain by simply changing the endpoint.
```

File: builders/interact/ethereum-api/libraries/web3js.md [hunk 5]
```diff
@@ -0,0 +1,498 @@
...
+```bash
+touch balances.js
+```
+
+Next, you will create the script for this file and complete the following steps:
+
+1. [Set up the Web3 provider](#setup-web3-with-tanssi)
+2. Define the `addressFrom` and `addressTo` variables
+3. Create the asynchronous `balances` function which wraps the `web3.eth.getBalance` method
+4. Use the `web3.eth.getBalance` function to fetch the balances for the `addressFrom` and `addressTo` addresses. You can also leverage the `web3.utils.fromWei` function to transform the balance into a more readable number in ETH
```
Comment (eshaben) [inline builders/interact/ethereum-api/libraries/web3js.md:80]: Do we want to use ETH here?

File: builders/interact/ethereum-api/libraries/web3js.md [hunk 6]
```diff
@@ -0,0 +1,498 @@
...
+node balances.js
+```
+
+If successful, the balances for the origin and receiving address will be displayed in your terminal in ETH.
+
+![Check balance Web3js](/images/builders/interact/ethereum-api/web3js/web3js-1.png)
+
+### Send Transaction Script {: #send-transaction-script }
+
+You'll only need one file for executing a transaction between accounts. For this example, you'll be transferring 1 UNIT token from an origin address (from which you hold the private key) to another address. To get started, you can create a `transaction.js` file by running:
```
Comment (eshaben) [inline builders/interact/ethereum-api/libraries/web3js.md:123]:
```suggestion
You'll only need one file to execute a transaction between accounts. For this example, you'll be transferring 1 UNIT token from an origin address (from which you hold the private key) to another address. To get started, you can create a `transaction.js` file by running:
```

File: builders/interact/ethereum-api/libraries/web3js.md [hunk 7]
```diff
@@ -0,0 +1,498 @@
...
+
+```bash
+touch transaction.js
+```
+
+Next, you will create the script for this file and complete the following steps:
+
+1. [Set up the Web3 provider](#setup-web3-with-tanssi)
+2. Define the `addressFrom`, including the `privateKey`, and the `addressTo` variables. The private key is required to create a wallet instance. **Note: This is for example purposes only. Never store your private keys in a JavaScript file**
+3. Create the asynchronous `send` function which wraps the transaction object and the sign and send transaction functions
```
Comment (eshaben) [inline builders/interact/ethereum-api/libraries/web3js.md:133]:
```suggestion
3. Create the asynchronous `send` function, which wraps the transaction object, and the sign and send transaction functions
```

File: builders/interact/ethereum-api/libraries/web3js.md [hunk 8]
```diff
@@ -0,0 +1,498 @@
...
+    --8<-- 'code/ethereum-api/web3js/transaction.js'
+    ```
+
+To run the script, you can run the following command in your terminal:
+
+```bash
+node transaction.js
+```
+
+If the transaction was succesful, in your terminal you'll see the transaction hash has been printed out.
```
Comment (eshaben) [inline builders/interact/ethereum-api/libraries/web3js.md:184]:
```suggestion
If the transaction was successful, in your terminal, you'll see the transaction hash has been printed out.
```

File: builders/interact/ethereum-api/libraries/web3js.md [hunk 9]
```diff
@@ -0,0 +1,498 @@
...
+node deploy.js
+```
+
+If successful, the contract's address will be displayed in the terminal.
+
+![Deploy Contract Web3js](/images/builders/interact/ethereum-api/web3js/web3js-3.png)
+
+### Read Contract Data (Call Methods) {: #read-contract-data }
+
+Call methods are the type of interaction that don't modify the contract's storage (change variables), meaning no transaction needs to be sent. They simply read various storage variables of the deployed contract.
```
Comment (eshaben) [inline builders/interact/ethereum-api/libraries/web3js.md:285]:
```suggestion
Call methods are the type of interactions that don't modify the contract's storage (change variables), meaning no transaction needs to be sent. They simply read various storage variables of the deployed contract.
```

File: builders/interact/ethereum-api/libraries/web3js.md [hunk 10]
```diff
@@ -0,0 +1,498 @@
...
+```
+
+Then you can take the following steps to create the script:
+
+1. Import the `abi` from the `compile.js` file
+2. [Set up the Web3 provider](#setup-web3-with-tanssi)
+3. Create the `contractAddress` variable using the address of the deployed contract
+4. Create an instance of the contract using the `web3.eth.Contract` function and passing in the `abi` and  `contractAddress`
+5. Create the asynchronous `get` function
+6. Use the contract instance to call one of the contract's methods and pass in any inputs if necessary. For this example, you will call the `number` method which doesn't require any inputs. You can use `await` which will return the value requested once the request promise resolves
```
Comment (eshaben) [inline builders/interact/ethereum-api/libraries/web3js.md:300]:
```suggestion
6. Use the contract instance to call one of the contract's methods and pass in any inputs if necessary. For this example, you will call the `number` method, which doesn't require any inputs. You can use `await`, which will return the value requested once the request promise resolves
```

File: builders/interact/ethereum-api/libraries/web3js.md [hunk 11]
```diff
@@ -0,0 +1,498 @@
...
+```
+
+If successful, the value will be displayed in the terminal.
+
+![Get contract variable value Web3js](/images/builders/interact/ethereum-api/web3js/web3js-4.png)
+
+
+### Interact with Contract (Send Methods) {: #interact-with-contract }
+
+Send methods are the type of interaction that modify the contract's storage (change variables), meaning a transaction needs to be signed and sent. In this section, you'll create two scripts: one to increment and one to reset the incrementer. To get started, you can create a file for each script and name them `increment.js` and `reset.js`:
```
Comment (eshaben) [inline builders/interact/ethereum-api/libraries/web3js.md:349]:
```suggestion
Send methods are the type of interactions that modify the contract's storage (change variables), meaning a transaction needs to be signed and sent. In this section, you'll create two scripts: one to increment and one to reset the incrementer. To get started, you can create a file for each script and name them `increment.js` and `reset.js`:
```

File: builders/interact/ethereum-api/libraries/web3js.md [hunk 12]
```diff
@@ -0,0 +1,498 @@
...
+    --8<-- 'code/ethereum-api/web3js/increment.js'
+    ```
+
+To run the script, you can enter the following command in your terminal:
+
+```bash
+node increment.js
+```
+
+If successful, the transaction hash will be displayed in the terminal. You can use the `get.js` script alongside the `increment.js` script to make sure that value is changing as expected:
```
Comment (eshaben) [inline builders/interact/ethereum-api/libraries/web3js.md:424]:
```suggestion
If successful, the transaction hash will be displayed in the terminal. You can use the `get.js` script alongside the `increment.js` script to make sure that the value is changing as expected.
```

File: builders/interact/ethereum-api/libraries/web3js.md [hunk 13]
```diff
@@ -0,0 +1,498 @@
...
+    --8<-- 'code/ethereum-api/web3js/reset.js'
+    ```
+
+To run the script, you can enter the following command in your terminal:
+
+```bash
+node reset.js
+```
+
+If successful, the transaction hash will be displayed in the terminal. You can use the `get.js` script alongside the `reset.js` script to make sure that value is changing as expected:
```
Comment (eshaben) [inline builders/interact/ethereum-api/libraries/web3js.md:494]:
```suggestion
If successful, the transaction hash will be displayed in the terminal. You can use the `get.js` script alongside the `reset.js` script to make sure that the value is changing as expected.
```

**Review by albertov19 — APPROVED** [review]
LGTM

**Review by eshaben — APPROVED** [review]

**Pull request merged**